### PR TITLE
rpc: small description/help improvements in blockchain.cpp

### PR DESCRIPTION
**Pull request by MarnixCroes**
To match the other descriptions/help doc
-start with Capital
-no `.` at the end
-`"Use coinstatsindex, if available."` -> `"Use coinstatsindex (if available)"`

### CODE REVIEW

**Comment by MarcoFalke** [comment]
If this is something that should be done, it should be enforced in the rpc help manager.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -297,7 +297,7 @@ static RPCHelpMan waitforblock()
                 "\nWaits for a specific new block and returns useful info about it.\n"
                 "\nReturns the current block on timeout or exit.\n",
                 {
-                    {"blockhash", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "Block hash to wait for."},
+                    {"blockhash", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "Block hash to wait for"},
                     {"timeout", RPCArg::Type::NUM, RPCArg::Default{0}, "Time in milliseconds to wait for a response. 0 indicates no timeout."},
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:301]: This one has a dot and above you are removing it.

**Review by maflcko — CHANGES_REQUESTED** [review]
I am not sure if this is worth it, given that it is *increasing* inconsistency?

**Comment by MarcoFalke** [comment]
Thank you for your contribution. While this stylistic change makes sense on its own, it comes at a cost and risk for the project as a whole. The weak motivation for the change does not justify the burden that it places on the project. A burden could be any of the following:

* Time spent on review
* Accidental introduction of bugs
* (Silent) merge conflicts, either in the branch or a backport branch. Those conflicts demand further developer and reviewer time or introduce bugs.

For more information about refactoring changes and stylistic cleanup, see
* https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#refactoring
* https://github.com/bitcoin/bitcoin/blob/master/.github/PULL_REQUEST_TEMPLATE.md
* https://github.com/bitcoin/bitcoin/issues/15465

Generally, if the style is not mentioned nor enforced by the developer notes, we leave it up to the original author to pick whatever fits them best personally and then leave it that way until the line is touched for other reasons.

Let me know if you have any questions.

**Pull request closed without merging**